baremetal/metal3machine_manager_test.go

```diff
@@ -2302,6 +2302,57 @@ var _ = Describe("Metal3Machine manager", func() {
 		}),
 	)
 
+	DescribeTable("Test updateObj function, original object shall remain unchanged",
+		func(Host *bmh.BareMetalHost) {
+			c := fakeclient.NewFakeClientWithScheme(setupSchemeMm(), Host)
+			machineMgr, err := NewMachineManager(c, nil, nil, nil, nil, klogr.New())
+
+			Expect(err).NotTo(HaveOccurred())
+
+			tcHost := Host.DeepCopyObject()
+			err = machineMgr.updateObject(context.TODO(), Host)
+
+			Expect(err).NotTo(HaveOccurred())
+			Expect(tcHost).To(Equal(Host))
+		},
+		Entry("updateObj host",
+			newBareMetalHost("myhost", nil, bmh.StateProvisioned, bmhPowerStatus(), true, true),
+		),
+		Entry("updateObj host next",
+			newBareMetalHost("myhost", nil, bmh.StateNone, bmhStatus(), false, false),
+		),
+	)
+
+	DescribeTable("Test createObj function",
+		func(BootstrapSecret *corev1.Secret) {
+			objects := []runtime.Object{}
+			c := fakeclient.NewFakeClientWithScheme(setupSchemeMm(), objects...)
+			machineMgr, err := NewMachineManager(c, nil, nil, nil, nil, klogr.New())
+
+			Expect(err).NotTo(HaveOccurred())
+
+			tcSecret := BootstrapSecret.DeepCopyObject()
+			err = machineMgr.createObject(context.TODO(), BootstrapSecret)
+
+			Expect(err).NotTo(HaveOccurred())
+			Expect(tcSecret).To(Equal(BootstrapSecret))
```

Comment:
Same concern here  as above. The secret should be fetched from the fakeclient IMO.

Reply:
I don't see your point. Test case checks basically that createObject doesn't tamper the BootstrapSecret in any way.

Reply:
I see, done.